pwiz/data/vendor_readers/Bruker/SpectrumList_Bruker.cpp

```diff
@@ -429,7 +441,41 @@ PWIZ_API_DECL SpectrumPtr SpectrumList_Bruker::spectrum(size_t index, DetailLeve
                 arrayType.units = MS_Vs_cm_2;
                 mobility->cvParams.emplace_back(arrayType);
 
-                spectrum->getCombinedSpectrumData(mz, intensity, mobility->data, config_.sortAndJitter);
+                if (msLevel > 1 && compassDataPtr_->isPassEntireDiaPasefFrame() && config_.includeIsolationArrays)
+                {
+                    // Each point has [m/z, intensity, IM, isoLow, isoHigh]
+                    BinaryDataArrayPtr isolationMzStart(new BinaryDataArray);
+                    result->binaryDataArrayPtrs.push_back(isolationMzStart);
+                    arrayType = MS_scanning_quadrupole_position_lower_bound_m_z_array;
+                    arrayType.units = MS_m_z;
+                    isolationMzStart->cvParams.emplace_back(arrayType);
+
+                    BinaryDataArrayPtr isolationMzEnd(new BinaryDataArray);
+                    result->binaryDataArrayPtrs.push_back(isolationMzEnd);
+                    arrayType = MS_scanning_quadrupole_position_upper_bound_m_z_array;
+                    arrayType.units = MS_m_z;
+                    isolationMzEnd->cvParams.emplace_back(arrayType);
+
+                    spectrum->getCombinedSpectrumData(mz, intensity, mobility->data, true, isolationMzStart->data, isolationMzEnd->data, config_.sortAndJitter);
+
+                    // Note the isolation range
+                    double isoLower = isolationMzStart->data[isolationMzStart->data.size() - 1];
+                    double isoUpper = isolationMzEnd->data[0];

```

Comment:
Why does "isoLower" get set to a value from the last window, but "isolUpper" gets set to a value from the first window?

Reply:
The values are presented in reverse sort order, just a quirk of the format